sklearn/model_selection/tests/test_search.py

```diff
@@ -1725,3 +1725,53 @@ def test_deprecated_grid_search_iid():
     grid = GridSearchCV(SVC(gamma='scale'), param_grid={'C': [1]}, cv=KFold(2))
     # no warning because no stratification and 54 % 2 == 0
     assert_no_warnings(grid.fit, X, y)
+
+
+def test_empty_cv_iterator_error():
+    # Generate sample data
```

Comment:
It seems you can simplify the tests by using the global X and y, and n_splits=3.